homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,87 @@ class SensorStateClass(StrEnum):
     SensorDeviceClass.WIND_SPEED: set(UnitOfSpeed),
 }
 
+DEVICE_CLASS_STATE_CLASSES: dict[SensorDeviceClass, set[SensorStateClass | None]] = {
+    SensorDeviceClass.APPARENT_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.AQI: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ATMOSPHERIC_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.BATTERY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO2: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CURRENT: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_RATE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_SIZE: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
```

Comment:
Same here, I think it might make sense to keep also MEASUREMENT (ie. I wish my bank account only ever went up)

Reply:
Bank measurement is a `total`, not `measurement`.

Reply:
I agree, this is something which makes sense to plot as either the state of the bank account, or the growth over a period.
I'm not sure `TOTAL_INCREASING` adds value though?

Reply:
Agree, lets remove `total_increasing` and see if the need shows up later maybe 👍